code/game/objects/items/weapons/teleprod.dm

```diff
@@ -0,0 +1,35 @@
+/obj/item/weapon/melee/baton/cattleprod/teleprod
+	name = "teleprod"
+	desc = "A prod with a bluespace crystal on the end. The crystal doesn't look too fun to touch."
+	icon_state = "teleprod_nocell"
+	item_state = "teleprod"
+	origin_tech = "combat=2;bluespace=4;materials=3"
+
+/obj/item/weapon/melee/baton/cattleprod/teleprod/attack(mob/living/carbon/M, mob/living/carbon/user)//handles making things teleport when hit
+	..()
+	if(status && (CLUMSY in user.mutations) && prob(50))
+		user.visible_message("<span class='danger'>[user] accidentally hits themself with [src]!</span>", \
+							"<span class='userdanger'>You accidentally hit yourself with [src]!</span>")
+		user.Weaken(stunforce*3)
+		deductcharge(hitcost)
+		do_teleport(user, get_turf(user), 50)//honk honk
+	else
+		if(status)
+			if(iscarbon(M) && !M.anchored)
+				do_teleport(M, get_turf(M), 15)
+
+/obj/item/weapon/melee/baton/cattleprod/attackby(obj/item/I, mob/user, params)//handles sticking a crystal onto a stunprod to make a teleprod
+	if(istype(I, /obj/item/weapon/ore/bluespace_crystal))
+		if(!bcell)
+			var/obj/item/weapon/melee/baton/cattleprod/teleprod/S = new /obj/item/weapon/melee/baton/cattleprod/teleprod
+			if(!remove_item_from_storage(user))
+				user.unEquip(src)
+			user.unEquip(I)
```

Comment:
`drop_item()` drops whatever is in the active hand by calling `unEquip(r_hand)` or `unEquip(l_hand)` as appropriate.